src/libraries/System.Private.CoreLib/src/Resources/Strings.resx

```diff
@@ -2653,6 +2653,9 @@
   <data name="IO_BindHandleFailed" xml:space="preserve">
     <value>BindHandle for ThreadPool failed on this handle.</value>
   </data>
+  <data name="IO_InconsistentLinkType" xml:space="preserve">
+    <value>The link's file system entry type is inconsistent with that of its target.</value>
```

Comment:
Should this include the path in question? We've been trying to include paths in exceptions where possible because it's so useful in logs.

Reply:
The request makes sense. I'll add it.

Reply:
I'm thinking I should add the path of the link, not the link of the target.

Reply:
Could this be a better message?
```
The type '{0}' cannot target the path '{1}' as it mismatches with the file system entry it represents.
```
And it would result in a message like this:
```
The type 'System.IO.DirectoryInfo' cannot target the path 'C:\User\david\Documents\report.pdf' as it mismatches with the file system entry it represents.

Reply:
That seems a little weird to me as I think we'd normally write it out like eg "Directory cannot have symbolic link to file '{0}'" and another string for file?

Reply:
I was thinking on passing the `Type` of the FIleSystemInfo we were going to return as an argument so it would be only one string. i.e:
```
SR.Format(SR.IO_InconsistentLinkType, typeof(DirectoryInfo), path)
```
Maybe it's too complicated and not worth it since it would require us to move the exception to the caller of this method in order to correctly pass the `Type` (given that someone could be using a custom class deriving from FileSystemInfo), but is do-able if we really want to be clear in the exception message (assuming my suggested message is clear).

Reply:
I think the latest version I pushed is clear enough.